packages/server/src/sdk/app/backups/imports.ts

```diff
@@ -0,0 +1,168 @@
+import { db as dbCore } from "@budibase/backend-core"
+import { TABLE_ROW_PREFIX } from "../../../db/utils"
+import { budibaseTempDir } from "../../../utilities/budibaseDir"
+import { DB_EXPORT_FILE, GLOBAL_DB_EXPORT_FILE } from "./constants"
+import {
+  uploadDirectory,
+  upload,
+} from "../../../utilities/fileSystem/utilities"
+import { downloadTemplate } from "../../../utilities/fileSystem"
+import { ObjectStoreBuckets, FieldTypes } from "../../../constants"
+import { join } from "path"
+import fs from "fs"
+import sdk from "../../"
+import { CouchFindOptions, RowAttachment } from "@budibase/types"
+const uuid = require("uuid/v4")
+const tar = require("tar")
+
+type TemplateType = {
+  file?: {
+    type: string
+    path: string
+  }
+  key?: string
+}
+
+async function updateAttachmentColumns(
+  prodAppId: string,
+  db: PouchDB.Database
+) {
+  // iterate through attachment documents and update them
+  const tables = await sdk.tables.getAllInternalTables(db)
+  for (let table of tables) {
+    const attachmentCols: string[] = []
+    for (let [key, column] of Object.entries(table.schema)) {
+      if (column.type === FieldTypes.ATTACHMENT) {
+        attachmentCols.push(key)
+      }
+    }
+    // no attachment columns, nothing to do
+    if (attachmentCols.length === 0) {
+      continue
+    }
+    // use the CouchDB Mango query API to lookup rows that have attachments
+    const params: CouchFindOptions = {
+      selector: {
+        _id: {
+          $regex: `^${TABLE_ROW_PREFIX}`,
+        },
+      },
+    }
+    attachmentCols.forEach(col => (params.selector[col] = { $exists: true }))
```

Comment:
Nice work getting this logic together - I'm thinking that when we start to look at "global media" we should rethink how URLs are stored, and probably not store them in the DB like this - possibly another part of using better bucket keys as well.

Reply:
Agreed, I would rather we didn't store URLs in the documents and instead just enrich something on the way out, as right now it means migrating them if the app ID changes etc etc.